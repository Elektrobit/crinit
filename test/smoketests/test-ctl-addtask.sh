# shellcheck shell=bash
#
# smoketest for crinit-ctl addtask command
#

setup() {
    crinit_config_setup
    sed "s#TASKDIR = .*#TASKDIR = ${SMOKETESTS_CONFDIR}#" < "${SMOKETESTS_CONFDIR}"/local.series > "${SMOKETESTS_CONFDIR}"/demo.series
    sed "s#TASKDIR = .*#TASKDIR = ${SMOKETESTS_CONFDIR}/addseries#" < "${SMOKETESTS_CONFDIR}"/addseries/add.series > "${SMOKETESTS_CONFDIR}"/addseries/demoadd.series
}

run() {
    crinit_daemon_start "${SMOKETESTS_CONFDIR}"/demo.series
    sleep 3

    # add a new task
    if ! "${BINDIR}"/crinit-ctl addtask "${SMOKETESTS_CONFDIR}"/sleep_one_day.crinit; then
        echo "crinit-ctl addtask failed unexpectedly"
        return 1
    fi

    # try to add it again, which should fail
    if "${BINDIR}"/crinit-ctl addtask "${SMOKETESTS_CONFDIR}"/sleep_one_day.crinit; then
        echo "crinit-ctl addtask should have failed"
        return 1
    fi

    if ! "$BINDIR"/crinit-ctl list | grep -q sleep_one_day; then
        echo "crinit-ctl addtask successful, but task not in list"
        return 1
    fi
}

teardown() {
    # Terminate crinit daemon
    crinit_daemon_stop
}

