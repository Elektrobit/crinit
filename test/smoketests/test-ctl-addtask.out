NAME                 PID  STATUS
hello_world           -1  done
sleep_one_sec         -1  done
after_sleep           -1  done
one_second_respawn    -1  loaded
fail_loop             -1  loaded
sleep_one_day       3547  running
